modules/overrides/user_configurable_overrides.go

```diff
@@ -94,7 +101,13 @@ func (o *userConfigurableOverridesManager) starting(ctx context.Context) error {
 		return errors.Wrap(err, "unable to start overrides subservices")
 	}
 
-	return o.reloadAllTenantLimits(ctx)
+	err := o.reloadAllTenantLimits(ctx)
+	if err != nil {
+		// Don't block start up if loading user-configurable overrides failed. We can fall back to runtime overrides.
+		metricUserConfigurableOverridesReloadFailed.Inc()
+		level.Error(o.logger).Log("msg", "failed to load user-configurable config during start-up, will fall back to runtime overrides", "err", err)
```

Comment:
I'm a bit torn here tbh.

Loading defaults would mean we use the overrides from the runtime configmap. In the user-configurable overrides you can enable and tweak features like the generic forwarders and the metrics-generators. So the most likely impact will be that the forwarder and metrics-generator for a tenant would be disabled again until we can reload these overrides.
The user-configurable overrides does not store operational limits btw (like max_live_traces), so these would still be respected.

Alternatively, if the component can not start up we might break ingestion for the entire cluster since this also runs on the distributors. This would mean that during an extended backend outage we can not restart any component using user-configurable overrides since they would fail to get past this check.

Reply:
Alternative idea: we can make it configurable. Default to crashing when loading fails but add a flag to skip verification.
-> In case of emergency, we can disable this and still load the app.

But I also feel conflicted about this:
- if we will want to disable this check during an emergency, why even bother crashing for it?
- it makes it explicit which is good but also causes more work

Reply:
As discussed offline we believe that failing to start if the user configurable overrides do not get loaded correctly is the correct path.

1) It mimics the behavior of loading the file based overrides
2) Its failure modes seem less dire.
3) It is more "future proof". Adding critical limits to user configurable overrides is better handled.

Reply:
I have reverted this change. We discussed this directly for a bit but basically: to keep Tempo simple to operate we'd favour an explicit in your face crash over a log line that warns you about something being broken. If the overrides don't work you might be running without overrides for a while which can also be disastrous as we add more responsibility to this module.
Disadvantage is that might block ingest into a Tempo cluster in very specific situations, for those scenarios we can consider adding a 'break glass' functionality that bypasses all startup checks.

Reply:
Oh, I didn't see your comment when I wrote this 😅